src/TestSuite/TestCase.php

```diff
@@ -88,6 +94,25 @@ public function skipIf($shouldSkip, $message = '')
     }
 
     /**
+     * update the ErrorLevel.
+     * @param bool $restore restore errorLevel if $restore = true.
+     * @return void
+     */
+    private static function _updateErrorLevel($restore = false)
+    {
+        $needle = 'Cake\Test\TestCase';
+        $class = get_called_class();
+        if (substr($class, 0, 18) !== $needle) {
+            return;
+        }
+        if ($restore) {
```

Comment:
Instead of a modal parameter it would be nice to have 2 methods as it will read better.

Reply:
In that case we need 3 method
1)check for tests is for core not user
2)getter
3)setter

Reply:
Could we update the teats that change error reporting so that they restore the old level?

Reply:
We no check changes of error reporting within testCases 
So if some of tests make change this level and if no restore the level to default (within that test) this cause change of error level in during of running tests(see output of my debug(error_reporting) in above
And after that some of test's error ignored 
We need save error level before the tests and restore it after running every test
We want to make an isolation test

Reply:
We must save/restore it even with using deprecated() helper 
We no a way to restore error level if we expect a exception for example `testAddInputTypeException`
resore line not execute anymore 
And we should  manage it with `setUp` `teardown`

Reply:
Yes. The tests that modify error_reporting() should be updated to restore the previous level. We have to do this for other process level state like the current locale.